analyzers/tests/SonarAnalyzer.UnitTest/Extensions/BlockSyntaxExtensionsTests.cs

```diff
@@ -0,0 +1,316 @@
+            block = block.WithOpenBraceToken(Token(SyntaxKind.OpenBraceToken).WithTrailingTrivia(ParseTrailingTrivia("// Comment")));
+            block.Should().BeNotEmpty(treatCommentsAsContent: true);
+        }
+
+        [TestMethod]
+        public void IsEmpty_LeadingTriviaOnCloseBrace_Comment()
+        {
+            var block = BlockWithoutTrivia();
+            block = block.WithCloseBraceToken(Token(SyntaxKind.CloseBraceToken).WithLeadingTrivia(ParseLeadingTrivia("// Comment")));
+            block.Should().BeNotEmpty(treatCommentsAsContent: true);
+        }
+
+        [TestMethod]
+        public void IsEmpty_TrailingTriviaOnOpenBrace_ConditionalCompilation()
+        {
+            var block = BlockWithoutTrivia();
+            block = block.WithOpenBraceToken(Token(SyntaxKind.OpenBraceToken).WithTrailingTrivia(ParseTrailingTrivia("#if Something\r\n var i = 1; \r\n #endif")));
+            block.Should().BeNotEmpty(treatConditionalCompilationAsContent: true);
+        }
+
+        [TestMethod]
+        public void IsEmpty_LeadingTriviaOnCloseBrace_ConditionalCompilation()
+        {
+            var block = BlockWithoutTrivia();
+            block = block.WithCloseBraceToken(Token(SyntaxKind.CloseBraceToken).WithLeadingTrivia(ParseLeadingTrivia("#if Something\r\n var i = 1; \r\n #endif")));
+            block.Should().BeNotEmpty(treatConditionalCompilationAsContent: true);
+        }
+
+        private static BlockSyntax BlockWithoutTrivia()
+        {
+            var block = MethodDeclarationForBlock("{}").Body;
+            block = block
+                .WithOpenBraceToken(block.OpenBraceToken.WithoutTrivia())
+                .WithCloseBraceToken(block.CloseBraceToken.WithoutTrivia());
+            return block;
+        }
+
+        private static MethodDeclarationSyntax MethodDeclarationForBlock(string methodBlock)
+            => MethodDeclaration(WrapInClass(methodBlock));
+
+        private static string WrapInClass(string methodBlockOrArrow) => $@"
+public class C
+{{
+    public void M()
+    {methodBlockOrArrow}
+}}";
+
+        private static MethodDeclarationSyntax MethodDeclaration(string source)
+        {
+            var root = CSharpSyntaxTree.ParseText(source).GetRoot();
+            root.ContainsDiagnostics.Should().BeFalse();
+            return root.DescendantNodes().OfType<MethodDeclarationSyntax>().Single();
+        }
+    }
+
+    internal static class BlockSyntaxExtensions
+    {
+        public class BlockSyntaxAssertions : ReferenceTypeAssertions<BlockSyntax, BlockSyntaxAssertions>
+        {
+            public BlockSyntaxAssertions(BlockSyntax block) : base(block)
+            {
+            }
+
+            protected override string Identifier => "BlockSyntax";
+
+            public AndConstraint<BlockSyntaxAssertions> BeEmpty(bool treatCommentsAsContent = true, bool treatConditionalCompilationAsContent = true, string because = "", params object[] becauseArgs)
```

Comment:
We are not using `string because = "", params object[] becauseArgs` in the callers and we can remove them and the associated code.